src/main/java/com/rocketden/main/game_object/Game.java

```diff
@@ -0,0 +1,30 @@
+package com.rocketden.main.game_object;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.Map;
+
+import com.rocketden.main.model.Room;
+
+@Getter
+@Setter
+public class Game {
+
+    private Room room;
+
+    private Map<String, Player> players;
+
+    private Timer timer;
+
+    private void endGame() {
+        try {
+            if (timer.isTimeUp()) {
+                // TODO: Send socket update to end the game, add results to db.
+            }
+        } catch(InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+    }
```

Comment:
Just a note: I think another option is if this logic is handled in a service class (or maybe a scheduled task that decrements the time remaining/checks if a game is finished every second), but we can prob leave that for another discussion

Reply:
Yes that sounds good as well, we'll discuss this more once it's actually needed / implemented fully.